akka-persistence-typed/src/test/scala/akka/persistence/typed/internal/RecoveryPermitterSpec.scala

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.typed.internal
+
+import akka.actor.PoisonPill
+import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.scaladsl.adapter.{ TypedActorRefOps, TypedActorSystemOps }
+import akka.actor.typed.{ ActorRef, Behavior, TypedAkkaSpecWithShutdown }
+import akka.persistence.Persistence
+import akka.persistence.RecoveryPermitter.{ RecoveryPermitGranted, RequestRecoveryPermit, ReturnRecoveryPermit }
+import akka.persistence.typed.scaladsl.PersistentBehaviors.CommandHandler
+import akka.persistence.typed.scaladsl.{ Effect, PersistentBehaviors }
+import akka.testkit.typed.scaladsl.{ ActorTestKit, TestProbe }
+import com.typesafe.config.{ Config, ConfigFactory }
+import org.scalatest.concurrent.Eventually
+
+import scala.concurrent.duration._
+import scala.util.control.NoStackTrace
+
+object RecoveryPermitterSpec {
+
+  class TestExc extends RuntimeException("simulated exc") with NoStackTrace
```

Comment:
our usual pattern for this is `TE` ("Test Exception") with a message "Boom!" :-)

Reply:
Fixed :)